source/shared/cpp/ObjectModel/TextBlock.cpp

```diff
@@ -165,3 +168,141 @@ void TextBlock::SetHorizontalAlignment(const HorizontalAlignment value)
     m_hAlignment = value;
 }
 
+std::string TextBlock::ParseISO8601UsingRegex() const
+{
+	std::regex pattern("\\{\\{(DATE|Date|TIME|Time)\\((\\d{4})-{1}(\\d{2})-{1}(\\d{2})T(\\d{2}):{1}(\\d{2}):{1}(\\d{2})(Z|([+-])(\\d{2}):{1}(\\d{2}))(\\)|,Short\\)|,SHORT\\)|,Long\\)|,LONG\\))\\}\\}");
+    std::smatch matches;
+    std::string text = m_text;
+    std::ostringstream parsedostr;
+    time_t offset = 0; 
+    bool isDate = false, isShort = true;
+    enum MatchIndex
+    {
+        IsDate = 1,
+        YEAR,
+        Month,
+        Day,
+        Hour,
+        Min,
+        Sec,
+        TimeZone,
+        TZHr = 10,
+        TZMn,
+        Format,
+    };
+    int factor = 1, hours = 0, minutes = 0;
+    struct tm parsedTm = { 0 };
+    int *addrs[] = {&parsedTm.tm_year, &parsedTm.tm_mon, 
+        &parsedTm.tm_mday, &parsedTm.tm_hour, &parsedTm.tm_min, 
+        &parsedTm.tm_sec, &hours, &minutes}; 
+    std::vector<int> indexer = {YEAR, Month, Day, Hour, Min, Sec, TZHr, TZMn};
+
+    while(std::regex_search(text, matches, pattern))
+    {
+        parsedostr << matches.prefix().str();
+
+        if(matches[IsDate].matched)
+        {
+            // Date is matched
+            isDate = (matches[IsDate].str()[0] == 'D')? true : false;
+        }
+
+        for(unsigned int idx = 0; idx < indexer.size(); idx++) 
+        { 
+            if(matches[indexer[idx]].matched)
+            {
+                // get indexes for time attributes to index into conrresponding matches
+                // and covert it to string
+                *addrs[idx] = stoi(matches[indexer[idx]]);
+            }
+        }
+
+        // maches offset sign, 
+        // Z == UTC, 
+        // + == time added from UTC
+        // - == time subtracted from UTC
+        if(matches[TimeZone].matched)
+        {
+            
+            char zone = matches[TimeZone].str()[0];
+            if(zone == 'Z')
+            {
+                factor = 0;
```

Comment:
>factor [](start = 16, length = 6)

Instead of having this "factor" variable, can you just calculate the offset inside these if statements?  I think that will be clearer.

Reply:
will do; thanks

---
In reply to: [149207223](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r149207223) [](ancestors = 149207223)

Reply:
Just the addition of minutes + seconds, right?  That's fine

---
In reply to: [149214345](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r149214345) [](ancestors = 149214345,149207223)